plugins/easyimage/plugin.js

```diff
@@ -239,15 +245,81 @@
 					loader.upload( widgetDef.uploadUrl, widgetDef.additionalRequestParameters );
 
 					fileTools.markElement( img, 'uploadeasyimage', loader.id );
-
-					fileTools.bindNotifications( editor, loader );
 				}
 			}
 
 			data.dataValue = temp.getHtml();
 		} );
 	}
 
+	// Extends given uploadWidget `definition` with an upload progress bar, added within wrapper.
+	function addUploadProgressBar( editor, definition ) {
+		definition.skipNotifications = true;
```

Comment:
Why not use `CKEDITOR.tools.object.merge`? I don't see so many benefits of mutating definition over returning new, extended one and the code would IMO a lot better.

Reply:
Don't remember what was my reasoning here. Isn't it that `object.merge` does not modify any object but returns new instead? So that we'll have quite a bit of fun to make it work.

Eventual alternative is `tools.extend`.

Reply:
Let's skip it now as we're going to rewrite it to `features` anyway.